extensions/amp-auto-ads/0.1/ad-network-config.js

```diff
@@ -75,28 +75,28 @@ export function getAdNetworkConfig(type, autoAmpAdsElement) {
  */
 class AdSenseNetworkConfig {
   /**
-   * @param {!Element} autoAmpAdsElement
+   * @param {!Window} win
+   * @param {!Object<string, string>} setupParams
    */
-  constructor(autoAmpAdsElement) {
-    this.autoAmpAdsElement_ = autoAmpAdsElement;
+  constructor(win, setupParams) {
+    this.win_ = win;
+    this.setupParams_ = setupParams;
   }
 
-  /**
-   * @param {!Window} win
-   */
-  isEnabled(win) {
-    const branch = getAdSenseAmpAutoAdsExpBranch(win);
+  /** @override */
+  isEnabled() {
+    const branch = getAdSenseAmpAutoAdsExpBranch(this.win_);
     return branch != AdSenseAmpAutoAdsHoldoutBranches.CONTROL;
   }
 
   /** @override */
   getConfigUrl() {
-    const docInfo = documentInfoForDoc(this.autoAmpAdsElement_);
+    const docInfo = documentInfoForDoc(this.win_.document);
```

Comment:
Document is actually incorrect here. It should be an ampdoc. You will need to re-thread the calls to constructor with `ampdoc` instance itself. Ampdoc itself has a property `ampdoc.win` which you can use to query `getAdSenseAmpAutoAdsExpBranch`, but it's probably also better to switch that method to use `ampdoc` as well. 

Why? To support shadow docs which AMP explicitly allows. With shadow docs `win.document.querySelector()` will simply not be able to penetrate the shadow and return what you are looking for.